sdk/spring/azure-spring-integration-servicebus/src/main/java/com/azure/spring/integration/servicebus/converter/ServiceBusMessageConverter.java

```diff
@@ -77,59 +83,95 @@ protected IMessage fromByte(byte[] payload) {
+        getStringHeader(headers, CORRELATION_ID).ifPresent(message::setCorrelationId);
+        getStringHeader(headers, TO).ifPresent(message::setTo);
+        getStringHeader(headers, LABEL).ifPresent(message::setLabel);
+        getStringHeader(headers, REPLY_TO_SESSION_ID).ifPresent(message::setReplyToSessionId);
+        getStringHeader(headers, PARTITION_KEY).ifPresent(message::setPartitionKey);
+        getStringHeader(headers, VIA_PARTITION_KEY).ifPresent(message::setViaPartitionKey);
+
+        headers.forEach((key, value) -> message.getProperties().put(key, value.toString()));
+    }
 
-        headers.forEach((key, value) -> serviceBusMessage.getProperties().put(key, value.toString()));
+    private Optional<String> getStringHeader(MessageHeaders springMessageHeaders, String key) {
+        return Optional.ofNullable(springMessageHeaders.get(key))
+                       .map(Object::toString)
+                       .filter(StringUtils::hasText);
     }
 
     @Override
-    protected Map<String, Object> buildCustomHeaders(IMessage serviceBusMessage) {
+    protected Map<String, Object> buildCustomHeaders(IMessage message) {
         Map<String, Object> headers = new HashMap<>();
 
-        if (StringUtils.hasText(serviceBusMessage.getMessageId())) {
-            headers.put(AzureHeaders.RAW_ID, serviceBusMessage.getMessageId());
-        }
-
-        if (StringUtils.hasText(serviceBusMessage.getContentType())) {
-            String contentType = serviceBusMessage.getContentType();
-            try {
-                MimeType mimeType = MimeType.valueOf(contentType);
-                headers.put(MessageHeaders.CONTENT_TYPE, mimeType.toString());
-            } catch (InvalidMimeTypeException e) {
-                LOG.warn("Invalid mimeType '{}' from service bus message.", contentType);
-            }
-        }
-
-        if (StringUtils.hasText(serviceBusMessage.getReplyTo())) {
-            headers.put(MessageHeaders.REPLY_CHANNEL, serviceBusMessage.getReplyTo());
-        }
-
-        headers.putAll(serviceBusMessage.getProperties());
+        // Spring MessageHeaders
+        Optional.ofNullable(message.getMessageId())
+                .filter(StringUtils::hasText)
+                .ifPresent(s -> headers.put(MessageHeaders.ID, s));
+        Optional.ofNullable(message.getContentType())
+                .filter(StringUtils::hasText)
+                .ifPresent(s -> headers.put(MessageHeaders.CONTENT_TYPE, s));
+        Optional.ofNullable(message.getReplyTo())
+                .filter(StringUtils::hasText)
+                .ifPresent(s -> headers.put(MessageHeaders.REPLY_CHANNEL, s));
+
+        // AzureHeaders.
+        // Does not have SCHEDULED_ENQUEUE_MESSAGE, because it's meaningless in receiver side.
+        Optional.ofNullable(message.getMessageId())
+                .filter(StringUtils::hasText)
+                .ifPresent(s -> headers.put(AzureHeaders.RAW_ID, s));
```

Comment:
shall this RAW_ID be deprecated? I'd think it would better we just leave it as it is?

Reply:
Yes, it's deprecated. I keep this logic for backward compatibility.